libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork+Salesforce.m

```diff
@@ -39,17 +39,18 @@ - (void)setDefaultConnectCommunityId:(NSString *)defaultConnectCommunityId {
 - (void)setupSalesforceObserver {
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(userAccountManagerDidChangeCurrentUser:)
-                                                 name:SFUserAccountManagerDidChangeCurrentUserNotification
+                                                 name:SFUserAccountManagerDidChangeUserDataNotification
                                                object:nil];
 }
 
 #pragma mark SFAuthenticationManagerDelegate
-
 - (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
-    if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
+    SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
+    if ([accountManager isKindOfClass:[SFUserAccountManager class]]
+            && (change & SFUserAccountChangeCurrentUser)) {
```

Comment:
This is adding dependencies on `currentUser` that do not otherwise exist.  `CSFNetwork` instances are siloed by user.  It shouldn't care about changes to any other user than the account that's configured for this instance.